internal/host/host_test.go

```diff
@@ -4519,3 +4520,270 @@ var _ = Describe("Rebooting day2", func() {
+				IPV6Addresses: []string{"::1"},
+			}}
+			inventoryObj.Interfaces = append(inventoryObj.Interfaces, args.Interfaces...)
+			inventoryStr, err := common.MarshalInventory(inventoryObj)
+			Expect(err).ToNot(HaveOccurred())
+			err = manager.UpdateInventory(ctx, &host.Host, inventoryStr)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Load the host from the database again and check that it has been renamed
+			// as expected:
+			host = hostutil.GetHostFromDB(hostID, infraEnvID, db)
+			Expect(host).ToNot(BeNil())
+			Expect(host.RequestedHostname).To(Equal(args.Expected))
+		},
+		Entry(
+			"Doesn't rename if already has acceptable name",
+			Args{
+				Discovered: "myhost",
+				Requested:  "",
+				Interfaces: []*models.Interface{
+					{
+						Name:          "eth0",
+						MacAddress:    "71:A0:A4:6F:BE:C8",
+						IPV4Addresses: []string{"192.168.0.1"},
+					},
+				},
+				Expected: "",
+			},
+		),
+		Entry(
+			"Doesn't rename if already renamed",
+			Args{
+				Discovered: "localhost",
+				Requested:  "myhost",
+				Interfaces: []*models.Interface{
+					{
+						Name:          "eth0",
+						MacAddress:    "71:A0:A4:6F:BE:C8",
+						IPV4Addresses: []string{"192.168.0.1"},
+					},
+				},
+				Expected: "myhost",
+			},
+		),
+		Entry(
+			"Replaces IPv4 localhost with MAC of first NIC",
+			Args{
+				Discovered: "localhost",
+				Requested:  "",
+				Interfaces: []*models.Interface{
+					{
+						Name:          "eth0",
+						MacAddress:    "71:A0:A4:6F:BE:C8",
+						IPV4Addresses: []string{"192.168.0.1"},
+					},
+				},
+				Expected: "71-a0-a4-6f-be-c8",
+			},
+		),
+		Entry(
+			"Replace IPv6 localhost with MAC of first NIC",
+			Args{
+				Discovered: "localhost6",
+				Requested:  "",
+				Interfaces: []*models.Interface{
+					{
+						Name:          "eth0",
+						MacAddress:    "71:A0:A4:6F:BE:C8",
+						IPV4Addresses: []string{"192.168.0.1"},
+					},
+				},
+				Expected: "71-a0-a4-6f-be-c8",
+			},
+		),
+		Entry(
+			"Ignores NIC with MAC but no IP address",
+			Args{
+				Discovered: "localhost",
+				Requested:  "",
+				Interfaces: []*models.Interface{
+					{
+						Name:          "eth0",
+						MacAddress:    "71:A0:A4:6F:BE:C8",
+						IPV4Addresses: []string{},
+					},
+					{
+						Name:          "eth1",
+						MacAddress:    "42:5a:90:c8:24:dc",
+						IPV4Addresses: []string{"192.168.0.1"},
+					},
+				},
+				Expected: "42-5a-90-c8-24-dc",
+			},
+		),
+		Entry(
+			"Ignores NIC with local IPv4 address",
+			Args{
+				Discovered: "localhost",
+				Requested:  "",
+				Interfaces: []*models.Interface{
+					{
+						Name:          "eth0",
+						MacAddress:    "71:A0:A4:6F:BE:C8",
+						IPV4Addresses: []string{"127.0.0.1"},
+					},
+					{
+						Name:          "eth1",
+						MacAddress:    "42:5a:90:c8:24:dc",
+						IPV4Addresses: []string{"192.168.0.1"},
+					},
+				},
+				Expected: "42-5a-90-c8-24-dc",
+			},
+		),
+		Entry(
+			"Ignores NIC with local IPv6 address",
```

Comment:
Typo(?): `loopback` instead of `local`. Local IP address is something different

Reply:
Done.